zebrad/src/components/sync/recent_sync_lengths.rs

```diff
@@ -35,13 +42,16 @@ impl RecentSyncLengths {
 
     /// Create a new instance of [`RecentSyncLengths`]
     /// and a [`watch::Receiver`] endpoint for receiving recent sync lengths.
-    pub fn new() -> (Self, watch::Receiver<Vec<usize>>) {
+    pub fn new(
+        progress_callback: Option<Arc<dyn Fn() + Send + Sync>>,
```

Comment:
This seem unnecessarily opaque, I think it would be better to pass the mutex itself.